testcontainers/src/core/env/config.rs

```diff
@@ -125,12 +128,58 @@ impl Config {
     ///  1. Docker host from the `tc.host` property in the `~/.testcontainers.properties` file.
     ///  2. `DOCKER_HOST` environment variable.
     ///  3. Docker host from the `docker.host` property in the `~/.testcontainers.properties` file.
-    ///  4. Else, the default Docker socket will be returned.
-    pub(crate) fn docker_host(&self) -> &str {
+    ///  4. Read the default Docker socket path, without the unix schema. E.g. `/var/run/docker.sock`.
+    ///  5. Read the rootless Docker socket path, checking in the following alternative locations:
+    ///     1. `${XDG_RUNTIME_DIR}/.docker/run/docker.sock`.
+    ///     2. `${HOME}/.docker/run/docker.sock`.
+    ///     3. `${HOME}/.docker/desktop/docker.sock`.
+    ///     4. `/run/user/${UID}/docker.sock`, where `${UID}` is the user ID of the current user.
+    ///  6. The default Docker socket including schema will be returned if none of the above are set.
+    pub(crate) fn docker_host(&self) -> Cow<'_, str> {
         self.tc_host
             .as_deref()
             .or(self.host.as_deref())
-            .unwrap_or(DEFAULT_DOCKER_HOST)
+            .map(Cow::Borrowed)
+            .unwrap_or_else(|| {
+                if cfg!(unix) {
+                    check_path_exists("/var/run/docker.sock".into())
+                        .or_else(|| {
+                            var("XDG_RUNTIME_DIR").ok().and_then(|dir| {
+                                check_path_exists(format!("{dir}/.docker/run/docker.sock",))
+                            })
+                        })
+                        .or_else(|| {
+                            check_path_exists(format!(
+                                "{}/.docker/run/docker.sock",
+                                var("HOME").unwrap()
+                            ))
+                        })
+                        .or_else(|| {
+                            check_path_exists(format!(
+                                "{}/.docker/desktop/docker.sock",
+                                var("HOME").unwrap()
+                            ))
+                        })
+                        .or_else(|| {
+                            // Get the user ID of the current user by running `id -u`
+                            StdCommand::new("id")
+                                .arg("-u")
+                                .output()
+                                .ok()
+                                .and_then(|output| {
+                                    from_utf8(&output.stdout).map(|s| s.trim().to_string()).ok()
+                                })
+                                .and_then(|uid| {
+                                    check_path_exists(format!("/run/user/${uid}/docker.sock"))
+                                })
+                        })
```

Comment:
We could consider to use [uzers crate](https://crates.io/crates/uzers) which wraps libc call.
Or even ignore this particular option, users can override it with config options 🤔 
I mean we can support this option on demand, for now we didn't have such requests.

However, we can keep this impl too

Reply:
I have removed this path fallback.